light_curves/code_src/data_structures.py

```diff
@@ -46,14 +46,20 @@ def append(self,x):
         
         Parameters
         ----------
-        x : Pandas dataframe
+        x : Pandas dataframe, optional
             contains columns [flux, fluxerr] and multi-index [objectid, label, band, time]
+            If x is None, calling this method has no affect (used when an archive call finds no light curve data).
         """
+        # if an archive call found no data, x is None. return without doing anything
+        if x is None:
+            return
```

Comment:
It may or may not be applicable (and if it's, it's a more substantial fix than this current PR): over the years in astroquery, we converged to the solution of returning empty Table-s (or whatever the expected valid return type is) when there is no match, rather than None or an error, so to avoid all downstream code to deal with these types of conditionals.

Reply:
Ah great, thanks for the heads up. The functions here are handling this case in different ways (not all of them use astroquery, and even the ones that do transform the data before returning and so may still handle this differently) so the idea here is to standardize to _something_. The consensus from those who voiced opinions previously was to use `None`, but if you have a strong opinion in favor of the astroquery method I could instead make these return empty dataframes to match the astroquery workflow.

Reply:
Nope, I don't have strong opinions, and since there aren't really downstream users here, the actual choice will matter less. It was just the experience that handling None/Error returns was more problematic at the end than returning empty containers. (E.g. it super easily distinguishes actual query errors/service outages/etc from the valid result of not finding matches, etc.)

Reply:
It was actually simpler to standardize to an empty dataframe so I just went with that. (More info in #138.)